### ISSUE TITLE
Sounds: allow for looping to be toggles in the sound inspector & using the API

### ISSUE THREAD

**Issue by xezno** [Facepunch/sbox-issues#1900]
### For?

S&Box

### What can't you do?

Making a looping sound is awkward. You have to use loop points / loop markers, which means using a separate program, which is dumb - it should just all be possible without having to leave the engine.

### How would you like it to work?

Should just be a checkbox to toggle looping on a sound in the inspector.
Could also have some way of selecting where the sound loops to/from in the inspector too, so that all functionality gets retained.

API-wise it would be nice to have just a simple `.SetLooping( bool isLoopingEnabled )` - then you can use it like this:
```cs
var sound = Sound.FromEntity( "radio.music_loop", this ).SetLooping( true );
```

### What have you tried?

You can already loop stuff using loop markers, which require separate software to add to a sound. 
There's no way to do any of this in the engine itself.

**Comment by gvarados1** [comment]
Yeah we used to have the ability to make looping sounds in-engine but that was (unintentionally?) remove when the old inspector tool was removed. Right now we have to use one of those weird thirdparty sound looping softwares.
